lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -439,15 +439,10 @@ def evaluate_selection_with_args(kwarg_arguments, field_defn, next_path, ast_nod
         end
 
         def dead_result?(selection_result)
-          r = selection_result
-          while r
-            if r.graphql_dead
-              return true
-            else
-              r = r.graphql_parent
-            end
-          end
-          false
+          # When a result is marked dead, any of its existing children are marked dead
+          # and it accepts no more writes. But this result might have been initialized,
+          # but not yet assigned to a key in its parent. So check the parent too.
+          selection_result.graphql_dead || ((parent = selection_result.graphql_parent) && parent.graphql_dead)
```

Comment:
can the second half of the clause just use `&.` or do you support ruby versions without that still?

Reply:
Yeah, technically the gem still supports Ruby 2.2, but I should probably _not_, since I think it's EOL: 

https://github.com/rmosolgo/graphql-ruby/blob/73d637d65aee3195f4e5c65a0118d94101bdee15/graphql.gemspec#L16

https://github.com/rmosolgo/graphql-ruby/blob/73d637d65aee3195f4e5c65a0118d94101bdee15/.rubocop.yml#L7